### PR TITLE
feat: even more metadata for opentelemetry

### PR DESCRIPTION
**Pull request by madwort**

### CODE REVIEW

**Review by bloodearnest — APPROVED** [review]

File: jobserver/api/jobs.py [hunk 1]
```diff
@@ -119,6 +119,9 @@ def post(self, request, *args, **kwargs):
 
                     span.set_attribute("db_id", str(job.id))
                     span.set_attribute("job_identifier", job.identifier)
+                    span.set_attribute("backend", job_request.backend)
```
Comment (ghickman) [inline jobserver/api/jobs.py:122]: This will be the related Backend object, is that expected?

File: jobserver/api/jobs.py [hunk 2]
```diff
@@ -119,6 +119,9 @@ def post(self, request, *args, **kwargs):
...
+                    span.set_attribute("created_by", job_request.created_by)
+                    span.set_attribute("workspace", job_request.workspace)
```
Comment (ghickman) [inline jobserver/api/jobs.py:124]: This will be the related Workspace object, is that expected?

File: jobserver/api/jobs.py (hunk 1 above)
Comment (madwort) [reply jobserver/api/jobs.py:122]: yeah, maybe I want `job_request.backend.slug`?

Comment (madwort) [reply jobserver/api/jobs.py:122]: (I guess I was hoping something along the line would convert this to a string for me)

Comment (madwort) [reply jobserver/api/jobs.py:122]: https://github.com/opensafely-core/job-server/pull/1717 perhaps?

**Pull request merged**